internal/db/schema/migrations/postgres/2/05_job.up.sql

```diff
@@ -0,0 +1,79 @@
+begin;
+
+create table job (
+    id text primary key,
+    name wt_name not null,
+    description wt_description not null,
+    code text not null
+        constraint job_code_must_be_not_empty
+            check(length(trim(code)) > 0),
+    next_scheduled_run timestamp with time zone not null
+        default current_timestamp,
+
+    constraint job_name_code_uq
+        unique(name, code)
+);
+
+comment on table job is
+    'job is a base table where each row represents a unique job that can only have one running instance at any specific time.';
+
+create table job_name_enm (
+    name text not null primary key
+);
+
+comment on table job_name_enm is
+    'job_name_enm is an enumeration table where each row contains the name of a valid job.';
+
+create table job_run (
+     id serial primary key,
+     job_id text not null
+         constraint job_fk
+             references job(id)
+             on delete cascade
+             on update cascade,
+     server_id text
+         constraint server_fk
+             references server(private_id)
+             on delete set null
+             on update cascade,
+     start_time timestamp with time zone not null
+         default current_timestamp,
+     end_time timestamp with time zone,
+     last_heartbeat timestamp with time zone not null
+         default current_timestamp,
+     completed_count int not null,
```

Comment:
It would be helpful and more descriptive to add a constraint that prevents `completed_count` from having negative numbers. The `job_run_completed_count_less_than_equal_to_total_count` will catch it but only because `total_count` cannot be 0.

Reply:
thks!